pkg/rancher-desktop/config/settings.ts

```diff
@@ -475,113 +487,162 @@ function updateSettings(settings: Settings) {
 // Setting IDs
 export const SETTING = { PL_RANCHER_VALUE: 'rancher' };
 
+const REGISTRY_PATH_PROFILE = ['SOFTWARE', 'Rancher Desktop', 'Profile'];
+
 /**
- * Read and validate the json formatted deployment profile.
- * Returns the system profile if it exists, otherwise the user profile if it exists.
- * @returns A settings profile stripped of all fields not found in the schema.
+ * Read and validate deployment profiles, giving system level profiles
+ * priority over user level profiles.  If the system directory contains a
+ * defaults or locked profile, the user directory will not be read.
+ * @returns type validated defaults and locked deployment profiles, and throws
+ *          an error if there is an error parsing the locked profile.
  */
-export function readDeploymentProfile() {
-  let profileData;
-  let profile;
+export function readDeploymentProfiles() {
+  let profiles = {
+    defaults: undefined,
+    locked:   undefined,
+  };
 
   switch (os.platform()) {
   case 'win32':
-    try {
-      profile = readRegistryDeploymentProfile(); // TODO: call with path to system profile
-    } catch {
+    for (const key of [nativeReg.HKLM, nativeReg.HKCU]) {
+      const registryKey = nativeReg.openKey(key, REGISTRY_PATH_PROFILE.join('\\'), nativeReg.Access.READ);
+
       try {
-        profile = readRegistryDeploymentProfile(); // TODO: call with path to user profile
-      } catch {}
+        if (registryKey !== null) {
+          profiles.defaults = readRegistryUsingSchema(defaultSettings, registryKey, ['Defaults']);
+          profiles.locked = readRegistryUsingSchema(defaultSettings, registryKey, ['Locked']);
+        }
+      } catch (err) {
+        console.error( `Error reading deployment profile: ${ err }`);
+      } finally {
+        nativeReg.closeKey(registryKey);
```

Comment:
This should be called only if `registryKey !== undefined`

Reply:
It's just fine to call closeKey with a null or undefined key, so I didn't want to clutter the code with the if, but the change has been made.